core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -163,6 +173,8 @@ final class ShuffleBlockFetcherIterator(
         case _ =>
       }
     }
+    freeMemory(getUsed)
+    shuffleFiles.foreach(_.delete())
```

Comment:
Since these are just temp files, we should not fail the job if file deletion failed. Let's try catch the deletion and log the exception instead of throwing it.